test/acceptance/features/immutable-ready-service-binding-resources.feature

```diff
@@ -0,0 +1,167 @@
+                application:
+                    name: app-immutable
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        When Service Binding "sbr-immutable" is ready
+        Then Immutable Service Binding is unable to be applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: sbr-immutable
+            spec:
+                application:
+                    name: app-immutable-2
+                    group: apps
+                    version: v1
+                    resource: deployments
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service-immutable
+            """
+
+    Scenario: Allow modifying a Service Binding when it is not yet ready due to "Application Not Found"
+        Given The Custom Resource is present
+        """
+        apiVersion: stable.example.com/v1
+        kind: Backend
+        metadata:
+            name: service1
+            annotations:
+                service.binding/host: path={.spec.host}
+        spec:
+            host: foo
+        """
+        And Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: sbr-1
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service1
+                application:
+                    name: app1
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        And jq ".status.conditions[] | select(.type=="Ready").status" of Service Binding "sbr-1" should be changed to "False"
+        And The application "app1" does not exist
+        When Generic test application "app2" is running
+        And Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: sbr-1
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service1
+                application:
+                    name: app2
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        Then Service Binding "sbr-1" is ready
+
+
+    Scenario: Allow modifying a Service Binding when it is not yet ready due to "Service Not Found"
+        Given Generic test application "app3" is running
+        And The service "service2" does not exist
```

Comment:
this step does not provide any added value, because clearly the service is not declared earlier. IMHO I would remove it.